pkg/monitoring/endpoint.go

```diff
@@ -0,0 +1,113 @@
+package monitoring
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/telekom/das-schiff-network-operator/pkg/frr"
+)
+
+const (
+	vrfAll = "all"
+)
+
+type Endpoint struct {
+	cli *frr.Cli
+}
+
+func NewEndpoint() *Endpoint {
+	return &Endpoint{cli: frr.NewCli()}
+}
+
+func writeResponse(data *[]byte, w http.ResponseWriter) {
+	_, err := w.Write(*data)
+	if err != nil {
+		http.Error(w, "failed to write response: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (e *Endpoint) ShowRoute(w http.ResponseWriter, r *http.Request) {
+	vrf := r.URL.Query().Get("vrf")
+	if vrf == "" {
+		vrf = vrfAll
+	}
+
+	protocol := r.URL.Query().Get("protocol")
+	if protocol == "" {
+		protocol = "ip"
+	}
```

Comment:
Please check if the protocol makes sense: ip and ipv6

Reply:
Done